src/Core.UnitTests/Notifications/NotificationServiceTests.cs

```diff
@@ -220,17 +220,43 @@ public void ShowNotification_UnknownInfoBarButtonClicked_NoException()
 
             infoBar.Raise(x => x.ButtonClick += null, new InfoBarButtonClickedEventArgs("unknown notification"));
 
+            callback.Invocations.Count.Should().Be(0);            
+        }
+
+        [TestMethod]
+        public void ShowNotification_InfoBarButtonClicked_ActionInvokedWithTheNotification_NotDismissed()
```

Comment:
This test is almost identical to the existing 'ShowNotification_InfoBarButtonClicked_ActionInvokedWithTheNotification_Dismissed`, isn't it?
i.e. the only difference is the check whether `DetachInfoBar` is null or not.

If that is the case, it would be simpler to have one data-driven test that accepts true/false and does the appropriate `DetachInfoBar` check.

(as it is, it isn't clear from the test names what the difference between the two tests is i.e. why we expect one to be dismissed and the other not).